ci/docker/python-wheel-windows-test-vs2017.dockerfile

```diff
@@ -0,0 +1,43 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# NOTE: You must update PYTHON_WHEEL_WINDOWS_IMAGE_REVISION in .env
+# when you update this file.
+
+# based on mcr.microsoft.com/windows/servercore:ltsc2019
+# contains choco and vs2017 preinstalled
+FROM abrarov/msvc-2017:2.11.0
+
+# Add unix tools to path
+RUN setx path "%path%;C:\Program Files\Git\usr\bin"
+
+# Remove previous installations of python from the base image
+# NOTE: a more recent base image (tried with 2.12.1) comes with python 3.9.7
+# and the msi installers are failing to remove pip and tcl/tk "products" making
+# the subsequent choco python installation step failing for installing python
+# version 3.9.* due to existing python version
```

Comment:
For the record, are there other upstream images that we could use?

Reply:
We could create our own images with Visual C++ based from the official Microsoft ones `FROM mcr.microsoft.com/windows/servercore:ltsc2019` example here: https://learn.microsoft.com/en-us/previous-versions/visualstudio/visual-studio-2017/install/build-tools-container?view=vs-2017 but I wanted to use the same one we have on the python-wheels for consistency: https://github.com/apache/arrow/blob/master/ci/docker/python-wheel-windows-vs2017.dockerfile#L23
There was someone working on updating the Visual C++ version to 2019, maybe I could try to create new Dockerfile images removing the `abrarov/msvc-2017` base image and create new ones based on the Microsoft ones `mcr.microsoft.com/windows/servercore`?
I would prefer to do that as a different ticket if you are ok with that.

Reply:
> [](https://github.com/apache/arrow/blob/master/ci/docker/python-wheel-windows-vs2017.dockerfile#L23)
> There was someone working on updating the Visual C++ version to 2019

That was me, but it was precisely stalled because of the "uninstall then install Python" problem.

Reply:
> I would prefer to do that as a different ticket if you are ok with that.

That's fine with me!